src/proof/nova.rs

```diff
@@ -3655,4 +3655,29 @@ pub mod tests {
             None,
         );
     }
+
+    #[test]
+    fn test_nested_letrec_regression() {
+        let s = &mut Store::<Fr>::default();
+        let expected = s.num(5);
+        let terminal = s.get_cont_terminal();
+        test_aux::<Coproc<Fr>>(
+            s,
+            "(letrec ((char2int (lambda (digit)
+                           (if (eq #\\0 digit) 0 1)))
+         (str2int (letrec ((inner (lambda (acc xs)
+                                          (if (eq xs \"\") acc
+                                              (inner (+ (* 2 acc)
+                                                        (char2int (car xs)))
+                                                     (cdr xs))))))
+                          (inner 0))))
+        (str2int \"101\"))",
+            Some(expected),
+            None,
+            Some(terminal),
+            None,
+            147,
+            None,
+        );
+    }
```

Comment:
Is there a somewhat more minimal example showing the issue?

Reply:
Hmm let me try to come up with one... It is a hard issue because `not_dummy` was incorrectly becoming false (thus concealing errors). I think I just need to wrap the let inside another let to show the issues listed above

Reply:
Okay I've come up with them:

1) `(letrec ((x (let ((z 0)) 1))) 2)` this fails because of point 1: the LETREC not_dummy is happening in the LET case because it checks for the NEXT continuation instead of the CURRENT, thus it tries to extend the environment with a dummy allocation
2) `(let ((x 0) (y x)) x)` this fails because of point 2 above, the use of `ConsName::Env`
3) `(letrec ((x 0) (y (letrec ((inner 1)) 2))) 0)` this fails because of of point 3 above: it is trying to follow two paths at once, the recursive env and nonrecursive env path. The reason it needs the inner letrec is to trick the not_dummy, as it checks.